sunpy/net/dataretriever/sources/goes.py

```diff
@@ -54,6 +57,24 @@ def _get_goes_sat_num(self, date):
             raise ValueError('No operational GOES satellites on {}'.format(
                 date.strftime(TIME_FORMAT)))
 
+    def _get_time_for_url(self, urls):
+        times = []
+        for uri in urls:
+            uripath = urlsplit(uri).path
+
+            # Extract the yymmdd or yyyymmdd timestamp
+            datestamp = os.path.splitext(os.path.split(uripath)[1])[0][4:]
```

Comment:
This should always be the case? Remove the first 4?

Reply:
yeah the filenames always start with `go{satnum:02d}` so always four char